src/clj/clojure/lang/map_entry.clj

```diff
@@ -20,6 +33,41 @@
     (list*
       'deftype type ['k 'v]
 
+      'ICounted
+      (list '-count ['this] 2)
+
+      'IIndexed
+      (list '-nth ['this 'n]
+        (list 'cond
+          (list 'zero? 'n) 'k
+          (list '= 'n 1) 'v
+          :else (list 'throw (list 'new-out-of-bounds-exception ""))))
+
+      (list '-nth ['this 'n 'not-found]
+        (list 'if (list 'or (list '= 'n 0) (list '= 'n 1))
+          (list '-nth 'this 'n)
+          'not-found))
+
+      'ILookup
+      (list '-lookup ['this 'k 'not-found]
+        (list 'if (list 'is-integer? 'k)
```

Comment:
Well, if it is in current Clojure, I can't really argue, but it doesn't seem a little odd to silently fail instead of raising an exception.